pkg/work/hub/controllers/manifestworkreplicasetcontroller/manifestworkreplicaset_deploy_reconcile.go

```diff
@@ -218,13 +218,22 @@ func (d *deployReconciler) clusterRolloutStatusFunc(clusterName string, manifest
 	// Get all relevant conditions
 	progressingCond := apimeta.FindStatusCondition(manifestWork.Status.Conditions, workv1.WorkProgressing)
 	degradedCond := apimeta.FindStatusCondition(manifestWork.Status.Conditions, workv1.WorkDegraded)
+	appliedCond := apimeta.FindStatusCondition(manifestWork.Status.Conditions, workv1.WorkApplied)
 
 	// Return ToApply if:
+	// - No Applied condition exists yet (work hasn't been applied by hub controller)
+	// - Applied condition hasn't observed the latest spec
 	// - No Progressing condition exists yet (work hasn't been reconciled by agent)
 	// - Progressing condition hasn't observed the latest spec
 	// - Degraded condition exists but hasn't observed the latest spec
 	//   (Degraded is optional, but if it exists, we wait for it to catch up)
-	if progressingCond == nil ||
+	//
+	// IMPORTANT: Check Applied condition FIRST to ensure the work has been properly applied
+	// before checking agent-side conditions. This prevents using stale timestamps from
+	// previous generations when conditions update their ObservedGeneration without changing Status.
+	if appliedCond == nil ||
```

Comment:
Is this the only change made to fix this [issue](https://github.com/open-cluster-management-io/ocm/issues/1237)?

Reply:
another fix is https://github.com/open-cluster-management-io/sdk-go/pull/163